client/src/components/LogInModal.jsx

```diff
@@ -0,0 +1,94 @@
+//Import necessary functionalities
+import React from 'react';
+import { UserStatus } from '../context/UserContext.jsx';
+import { useNavigate } from 'react-router-dom';
+
+const LogInModal = ({ isOpen, onClose }) => {
+
+    if (!isOpen) return null;
+
+    //Import display and user settings to use in display control and user updates
+    const { loggedUser, setLoggedUser } = UserStatus();
+    
+    //Call useNavigate hook to change component display upon log in
+    const navigate = useNavigate();
+
+    const handleSubmit = async (event) => {
+        event.preventDefault();
+        
+        //Check that correct value is being submitted right now
+        console.log(event.currentTarget.username.value);
+
+        //Set the value of the username input field to be stored in username variable
+        const username = event.currentTarget.username.value;
+
+        //Future Plans: Add in password functionality - OAuth?
+
+        //If on submission, there is a username submitted, send it to the server to check its existence
+        if (username) {
+            try {
+                const response = await fetch('/api/users/', {
+                    method: 'POST',
+                    headers: {
+                        'Content-Type': 'application/json',
+                    },
+                    body: JSON.stringify({ username })
+                });
+
+                if (!response.ok) {
+                    throw new Error("Failed to check log in");
```

Comment:
For your future work, when you write your tests, definitely test the happy path and not happy paths. So login works and when it fails at various levels.

Reply:
Great idea, thank you! I hadn't though to test error handling. I added this note as a [comment](https://tatig.atlassian.net/browse/SCRUM-142?focusedCommentId=10017) in the [SCRUM-142](https://tatig.atlassian.net/browse/SCRUM-142) task that's for testing later this week.